lib/ui/page/home/page/my_profile/crop_avatar/controller.dart

```diff
@@ -86,21 +86,51 @@ class CropController extends GetxController {
         frame.image.width.toDouble(),
         frame.image.height.toDouble(),
       );
-    } catch (e) {
-      if (e.toString().contains('Invalid image data') ||
-          e.toString().contains('The source image cannot be decoded') ||
-          e.toString().contains('Failed to detect image file format')) {
-        svg = await vg.loadPicture(
+    } catch (rasterError, rasterStack) {
+      try {
+        final PictureInfo picture = await vg.loadPicture(
           SvgStringLoader(String.fromCharCodes(image)),
           null,
         );
-
-        if (svg != null) {
-          dimensions.value = svg!.size;
-        }
-      } else {
-        rethrow;
+        dimensions.value = picture.size;
+        svg = picture;
+      } catch (svgError, svgStack) {
+        throw CombinedImageDecodeException(
+          rasterError: rasterError,
+          rasterStack: rasterStack,
+          svgError: svgError,
+          svgStack: svgStack,
+        );
       }
     }
   }
 }
+
+/// Exception thrown when both raster and SVG image decoding fails.
+class CombinedImageDecodeException implements Exception {
+  const CombinedImageDecodeException({
+    required this.rasterError,
+    required this.rasterStack,
+    required this.svgError,
+    required this.svgStack,
```

Comment:
Since the first group is `raster`, shouldn't the second be `vector`? Or otherwise the first one ideally shouldn't be `raster`, perhaps a `jpg` instead? `VectorGraphicUtilities` from `vector_graphics` package might support more than just SVGs, I guess.